ledger-api/grpc-definitions/com/daml/ledger/api/v1/active_contracts_service.proto

```diff
@@ -56,9 +58,59 @@ message GetActiveContractsResponse {
 
   // The workflow that created the contracts.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional, In the Ledger API 3.0 this field will be removed.
   string workflow_id = 2;
 
   // The list of contracts that were introduced by the workflow with ``workflow_id`` at the offset.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // In the Ledger API 3.0 this field will be removed.
+  // This is empty for multi-domain API.
   repeated CreatedEvent active_contracts = 3;
+
+  // Non-empty for multi-domain API.
+  // For each contract there could be multiple ContractStateComponent-s in the entire snapshot. These together define
+  // the state of the contract in the snapshot.
+  repeated ContractStateComponent contract_state_components = 4;
+}
+
+message ContractStateComponent {
+  oneof contract_state_component {
+    // The contract is active in the given domain.
+    // There will be one such message for each domain the contract is active in.
+    ActiveContract active_contract = 1;
+
+    // This message is only available, if the corresponding transfer-in event cannot be found before or at the
+    // active_at_offset, and at least one of the querying parties could submit the corresponding transfer-in command.
+    // The incomplete_transferred_out.created_event.witness_parties will contain all such querying parties.
+    IncompleteTransferredOut incomplete_transferred_out = 2;
+
+    // Important: this message is not indicating that the contract is active for the target domain!
+    // This message is only available, if the corresponding transfer-out event cannot be found before or at the
+    // active_at_offset, and at least one of the querying parties could have submitted the corresponding transfer-in command.
+    // The incomplete_transferred_in.created_event.witness_parties will contain all such querying parties.
+    TransferredInEvent incomplete_transferred_in = 3;
```

Comment:
Should we wrap this into a IncompleteTransferredIn ? That would bring nice symmetry (happy me) and allow for future backward compatible changes.